pkg/cloudevents/transport/nats/options.go

```diff
@@ -10,3 +14,15 @@ func WithEncoding(encoding Encoding) Option {
 		return nil
 	}
 }
+
+// WithConnOptions supplies NATS connection options that will be used when setting
+// up the internal NATS connection
+func WithConnOptions(opts ...nats.Option) Option {
```

Comment:
Just WithOptions imo.

Reply:
There are Nats related options and transport related options like the encoding so dropping the `Conn` might be confusing?

Reply:
But these are specific to NATS yes?

Reply:
no, in the same set of options we have `WithEncoding` which is a transport thing so `WithOptions` would be confusing, `WithConnOptions` makes it clear the ones passed there related to nats when scanning docs? and more might come in time thats unrelated to nats - since ultimately this config section configures the transport first

Reply:
ok fair.